src/bootstrap/check.rs

```diff
@@ -184,8 +195,8 @@ macro_rules! tool_check_step {
     };
 }
 
-tool_check_step!(Rustdoc, "src/tools/rustdoc");
-tool_check_step!(Clippy, "src/tools/clippy");
+tool_check_step!(Rustdoc, "src/tools/rustdoc", SourceType::InTree);
+tool_check_step!(Clippy, "src/tools/clippy", SourceType::Submodule);
```

Comment:
Clippy should be considersed `InTree` now. All the changes which break it are fixed directly in the Rust repository.
```suggestion
tool_check_step!(Clippy, "src/tools/clippy", SourceType::InTree);
```

Reply:
Clippy doesn't use the same lints. In particular, it fails the `unused_lifetimes` lint.

Reply:
Interesting.
cc @flip1995, probably something that should be fixed.

Reply:
Huh? Clippy is checked with `deny(warnings)` in CI, so it shouldn't fail any lints.

Reply:
`unused_lifetimes` is allow-by-default.  Bootstrap selects particular lints to enable [here](https://github.com/rust-lang/rust/blob/1fb612bd15bb3ef098fd24c20d0727de573b4410/src/bootstrap/builder.rs#L1097-L1119).

Reply:
Oh right, `unused_lifetimes` is allow-by-default. Where can we find the set of lints enabled for in-tree tools? I think Clippy has no problem in adapting those.

Reply:
See the link in my previous message.

Reply:
I opened rust-lang/rust-clippy#5713 to catch up with bootstrap. I wouldn't block this PR on that though. I can change Clippy back to `InTree` after/in the next `subtree pull`.